chsdi/lib/validation/identify.py

```diff
@@ -179,12 +179,29 @@ def tolerance(self, value):
     @timeInstant.setter
     def timeInstant(self, value):
         if value is not None:
-            if len(value) != 4:
-                raise HTTPBadRequest('Only years are supported as timeInstant parameter')
-            if value.isdigit():
-                self._timeInstant = int(value)
+            instants = []
+            timeInstants = value.split(',')
+            nbTimeInstants = len(timeInstants)
+            nbLayers = len(self.layers)
+            if nbTimeInstants == nbLayers:
+                for instant in timeInstants:
+                    if len(instant) not in (0, 4):
+                        raise HTTPBadRequest(
+                            'Only years are supported as timeInstant parameter: provided value %s' % instant)
+                    if instant.isdigit():
+                        instant = int(instant)
+                    elif len(instant) == 0:
+                        instant = None
+                    else:
+                        raise HTTPBadRequest(
+                            'Please provide an integer for the parameter timeInstant: provided value %s' % instant)
+                    instants.append(instant)
+            elif nbTimeInstants == 1:
```

Comment:
You miss the checks here in this code branch. (test taht the timeinstant is of correct lenght, is a digit, etc).